src/pytest_dbfixtures/factories/postgresql.py

```diff
@@ -121,7 +121,7 @@ def drop_postgresql_database(psycopg2, user, host, port, db):
     conn = psycopg2.connect(user=user, host=host, port=port)
     conn.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
     cur = conn.cursor()
-    cur.execute('DROP DATABASE IF EXISTS %s' % db)
+    cur.execute('DROP DATABASE IF EXISTS %s;' % db)
```

Comment:
Fair enough. The postgres fixture is unusable for me without the semicolon
since the db isn't cleared. You're saying the fixture works for you without
the semicolon?
On 10 Sep 2016 12:11, "Michał Masłowski" <notifications@github.com> wrote:

> In src/pytest_dbfixtures/factories/postgresql.py
> <https://github.com/ClearcodeHQ/pytest-dbfixtures/pull/185#discussion_r78275121>
> :
>
> > @@ -121,7 +121,7 @@ def drop_postgresql_database(psycopg2, user, host, port, db):
> >      conn = psycopg2.connect(user=user, host=host, port=port)
> >      conn.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
> >      cur = conn.cursor()
> > -    cur.execute('DROP DATABASE IF EXISTS %s' % db)
> > +    cur.execute('DROP DATABASE IF EXISTS %s;' % db)
>
> We have cur.execute('CREATE DATABASE ' + db). Also the DROP DATABASE
> works (I do get errors from it if a buggy test leaves an open database
> connection). I cannot find explicit rules about semicolons, but it seems
> the single statement passed via cur.execute doesn't need a trailing one.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ClearcodeHQ/pytest-dbfixtures/pull/185/files/492e2c8e62eac3b385abc095bd46bf2d08c93f9d#r78275121>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADHzHpb3TlFNnWr09vJuzjxfvzUGNtzXks5qooJggaJpZM4J5j-u>
> .
>